Code/Framework/AzCore/AzCore/Name/NameDictionary.cpp

```diff
@@ -120,6 +140,56 @@ namespace AZ
         return Name();
     }
 
+    void NameDictionary::LoadLiteral(Name& nameLiteral)
+    {
+        if (nameLiteral.m_data == nullptr)
+        {
+            Name nameData = MakeName(nameLiteral.m_view);
+            nameLiteral.m_data = AZStd::move(nameData.m_data);
+            nameLiteral.m_hash = nameData.m_hash;
+        }
+    }
+
+    void NameDictionary::LoadDeferredNames(Name* deferredHead)
+    {
+        if (deferredHead == nullptr)
+        {
+            return;
+        }
+
+        Name* current = deferredHead;
+        while (current != nullptr)
+        {
+            LoadLiteral(*deferredHead);
+            current = current->m_nextName;
+        }
+
+        if (!deferredHead->m_linkedToDictionary)
+        {
+            deferredHead->m_linkedToDictionary = true;
+            deferredHead->LinkStaticName(&m_deferredHead);
+        }
+    }
+
+    void NameDictionary::UnregisterDeferredName(Name& name)
+    {
+        if (m_deferredHead == &name)
+        {
+            m_deferredHead = name.m_nextName;
+        }
+    }
```

Comment:
This function is really unregistering a deferred head and not any type of deferred name.

Reply:
Yep good callout, updated name